packages/main/src/plugin/documentation/documentation-service.ts

```diff
@@ -0,0 +1,247 @@
+      } else {
+        throw new Error('Failed to fetch documentation JSON files');
+      }
+      this.isInitialized = true;
+    } catch (error) {
+      console.error('Failed to fetch documentation at startup:', error);
+      // Fallback to predefined documentation if fetching fails
+      this.documentation = this.getFallbackDocumentation();
+      this.isInitialized = true;
+    }
+  }
+
+  async getDocumentationItems(): Promise<DocumentationInfo[]> {
+    if (!this.isInitialized) {
+      await this.fetchDocumentation();
+    }
+    return this.documentation;
+  }
+
+  async refreshDocumentation(): Promise<void> {
+    this.isInitialized = false; // Force re-fetch
+    await this.fetchDocumentation();
+    this.apiSender.send('documentation-updated');
+  }
+
+  private async fetchJsonContent(url: string): Promise<Array<{ name: string; url: string }> | null> {
+    try {
+      const response = await fetch(url);
+
+      if (!response.ok) {
+        throw new Error(`HTTP ${response.status}: ${response.statusText}`);
+      }
+
+      return await response.json();
+    } catch (error) {
+      if (error instanceof Error) {
+        if (error.name === 'AbortError') {
+          throw new Error(`Request timeout while fetching ${url}`);
+        }
+        throw new Error(`Failed to fetch ${url}: ${error.message}`);
+      }
+      throw new Error(`Failed to fetch ${url}: Unknown error`);
+    }
+  }
+
+  private parseDocumentationFromJson(
+    docsJson: DocumentationJsonInfo[],
+    tutorialsJson: DocumentationJsonInfo[],
+  ): DocumentationInfo[] {
+    const documentation: DocumentationInfo[] = [];
+
+    // Validate input parameters
+    if (!docsJson || !tutorialsJson) {
+      console.warn('Missing JSON content for parsing documentation');
+      return this.getCoreDocumentationPages();
+    }
+
+    // Parse both docs and tutorials using generic logic
+    const parseConfigs = [
+      {
+        data: docsJson,
+        category: 'Documentation',
+        errorMessage: 'Error parsing documentation JSON:',
+      },
+      {
+        data: tutorialsJson,
+        category: 'Tutorial',
+        errorMessage: 'Error parsing tutorials JSON:',
+      },
+    ];
+
+    for (const config of parseConfigs) {
+      try {
+        for (const item of config.data) {
+          if (item.name && item.url) {
+            documentation.push({
+              id: `${config.category}-${this.generateId(item.name)}`,
+              title: item.name,
+              description: `${config.category}: ${item.name}`,
+              url: item.url,
+              category: config.category,
+            });
+          }
+        }
+      } catch (error) {
+        console.error(config.errorMessage, error);
+      }
+    }
+
+    // Remove duplicates and add core pages
+    const corePages = this.getCoreDocumentationPages();
+    return [...corePages, ...this.removeDuplicates(documentation)];
+  }
+
+  private getCoreDocumentationPages(): DocumentationInfo[] {
```

Comment:
Please move this to a dedicated asset file somewhere, I would rather not have hardcoded documention info in the typescript files